packages/block-library/src/post-featured-image/dimension-controls.js

```diff
@@ -72,53 +66,80 @@ const DimensionControls = ( {
 	};
 	const scaleLabel = _x( 'Scale', 'Image scaling options' );
 	return (
-		<PanelBody title={ __( 'Dimensions' ) }>
-			<Flex
-				justify="space-between"
-				className={ classNames(
-					'block-library-post-featured-image-dimension-controls',
-					{ 'scale-control-is-visible': !! height }
-				) }
+		<InspectorControls __experimentalGroup="dimensions">
+			<ToolsPanelItem
+				className="single-column"
+				hasValue={ () => !! height }
+				label={ __( 'Height' ) }
+				onDeselect={ () => setAttributes( { height: undefined } ) }
+				resetAllFilter={ () => ( {
+					height: undefined,
+				} ) }
+				isShownByDefault={ true }
+				panelId={ clientId }
+			>
+				<UnitControl
+					label={ __( 'Height' ) }
+					labelPosition="top"
+					value={ height || '' }
+					onChange={ ( nextHeight ) =>
+						onDimensionChange( 'height', nextHeight )
+					}
+					units={ units }
+				/>
+			</ToolsPanelItem>
+			<ToolsPanelItem
+				className="single-column"
+				hasValue={ () => !! width }
+				label={ __( 'Width' ) }
+				onDeselect={ () => setAttributes( { width: undefined } ) }
+				resetAllFilter={ () => ( {
+					width: undefined,
+				} ) }
+				isShownByDefault={ true }
+				panelId={ clientId }
 			>
-				<FlexItem>
-					<UnitControl
-						label={ __( 'Height' ) }
-						labelPosition="top"
-						value={ height || '' }
-						onChange={ ( nextHeight ) => {
-							onDimensionChange( 'height', nextHeight );
-						} }
-						units={ units }
-					/>
-				</FlexItem>
-				<FlexItem>
-					<UnitControl
-						label={ __( 'Width' ) }
-						labelPosition="top"
-						value={ width || '' }
-						onChange={ ( nextWidth ) => {
-							onDimensionChange( 'width', nextWidth );
-						} }
-						units={ units }
-					/>
-				</FlexItem>
-			</Flex>
+				<UnitControl
+					label={ __( 'Width' ) }
+					labelPosition="top"
+					value={ width || '' }
+					onChange={ ( nextWidth ) =>
+						onDimensionChange( 'width', nextWidth )
+					}
+					units={ units }
+				/>
+			</ToolsPanelItem>
 			{ !! height && (
-				<ToggleGroupControl
+				<ToolsPanelItem
```

Comment:
I'm also seeing the vanishing controls:


![Nov-17-2021 08-10-44](https://user-images.githubusercontent.com/6458278/142066345-9667781a-58f2-4693-b39f-6b01b5236b8c.gif)

I'm not 100% certain, but it seems to be related to the conditional rendering of the `ToolsPanel`. 

When I remove the wrapper, things work okay:

```jsx
			{ !! height && (
				<ToggleGroupControl
					label={ scaleLabel }
					value={ scale }
					help={ scaleHelp[ scale ] }
					onChange={ ( value ) =>
						setAttributes( {
							scale: value,
						} )
					}
					isBlock
				>
					{ SCALE_OPTIONS }
				</ToggleGroupControl>
			) }
```

Relocating the `height` condition also seems to work:

```jsx
			<ToolsPanelItem
                             ...
			>
				{ !! height && (
					<ToggleGroupControl
						label={ scaleLabel }
						value={ scale }
						help={ scaleHelp[ scale ] }
						onChange={ ( value ) =>
							setAttributes( {
								scale: value,
							} )
						}
						isBlock
					>
						{ SCALE_OPTIONS }
					</ToggleGroupControl>
				) }
			</ToolsPanelItem>
```

Though the above causes a `Scale` option to the appear in the ToolsPanel options menu even where the control isn't shown.

Do we need the scale control to appear in the ToolsPanel at all? 

It's "toggleability"&trade; is determined only by the `height` value as far as I can tell, so in my mind. Maybe we could reset the scale elsewhere, e.g., by detecting changes to the height if need be? 🤷

Reply:
Whoa, interesting find! Weird -- I know we have conditionally rendered ToolsPanelItems, like the way we only render enabled controls in the [typography panel](https://github.com/WordPress/gutenberg/blob/f323d9620845068751ad60742fc16e0e8fc1e76b/packages/block-editor/src/hooks/typography.js#L152). Nice spot, I'll play around with this 💯

Reply:
It feels a _bit_ off to remove the ToolsPanelItem wrapper, since then we'd have a control in the ToolsPanel which isn't tracked in the menu. I see your point here, though, since its reset is meant to be tightly linked to Height. Unfortunately it also seems to break some of the styling with the single-column height 🤔 If that's the way we go, I can fix that specifically for this block I think.

At the same time it also feels wrong to keep the ToolsPanelItem wrapper and only conditionally render the control for the reason you pointed out -- the option appears in the reset menu even when the control isn't shown.

Reply:
> since then we'd have a control in the ToolsPanel which isn't tracked in the menu

Yeah, I see what you mean.

I suppose it depends on what folks expect ToolsPanel options menu items to represent. 

For example, when controls are _not_ shown by default, we can toggle their visibility/reset their values independent of any sibling control. At least as far as I know!

This wouldn't apply to the Scale control where there is no height value.

The value I see in adding it to the menu is to indicate to users that resetting all, will also reset the scale value (which appears to occur anyway). But resetting all also makes that item disappear from the menu itself.

 It does introduce a new paradigm. 🤔